dream_demo/bin/dog.ml

```diff
@@ -0,0 +1,8 @@
+open Ppx_yojson_conv_lib.Yojson_conv.Primitives
+
+type t = { id : string; name : string; breed : string }
+[@@deriving yojson]
+
+let breed = Option.fold ~some:(fun dog -> dog.breed) ~none:""
```

Comment:
About the `breed` and `name` functions: I would expect `Dog.breed` to take `t` rather than `t option` (same with `name`).  May be a good idea to name these a bit differently to signal it takes `t option`. Potentially `breed'`.

Reply:
I renamed lots of things to end in `_opt`. Hopefully it is more clear now.